plugins/modules/sysctl.py

```diff
@@ -187,6 +213,16 @@ def process(self):
 
         # Do the work
         if not self.module.check_mode:
+            # Set file permissions if there are differences.
+            # - Don't replace self.changed directly it's not only for fs attr but also sysctl param changes.
+            # - Ansible 2.9.x does not support 'path' like load_file_common_arguments (params, path=PATH).
+            #   so set 'sysctl_file' as 'path' in module.params.
+            self.module.params['path'] = self.sysctl_file
+            file_args = self.module.load_file_common_arguments(self.module.params)
+            fs_attr_changed = self.module.set_fs_attributes_if_different(file_args, False)
+
+            if fs_attr_changed:
+                self.changed = True
```

Comment:
This is the way it works
```suggestion
            self.changed = self.module.set_fs_attributes_if_different(file_args, self.changed)

```

Reply:
Thank you for your detailed instruction! I have fixed it.
And I'm going to re-read the code and document links that you gave me before. thanks again!